node/src/keyshare/mod.rs

```diff
@@ -1,52 +1,46 @@
 pub mod gcp;
 pub mod local;
 
-use cait_sith::KeygenOutput;
-use k256::{AffinePoint, Scalar, Secp256k1};
-use serde::{Deserialize, Serialize};
+use crate::signature_provider::KeyshareId;
+use serde::{de, Deserialize, Serialize};
 
 /// The root keyshare data along with an epoch. The epoch is incremented
 /// for each key resharing. This is the format stored in the old MPC
 /// implementation, and we're keeping it the same to ease migration.
 #[derive(Clone, Serialize, Deserialize)]
-pub struct RootKeyshareData {
+pub struct RootKeyshareData<T: Clone + KeyshareId> {
```

Comment:
I think we need compatibility with existing keyshares stored on disk or on gcp:
 - The file path is different in the new implementation
 - The serde format is different - we used to store it as flat 3 members, but now it's nested.

I would suggest doing an implicit migration; i.e. on startup, if we discover that the new format doesn't exist but the old does, then create the new one.

For gcp, I don't know how that's going to work. We don't get access to create a separate secret for gcp, so I think we're going to have to be creative here and invent another encoded format that can store multiple keys.

Unfortunately, we do not have any migrate-ability (or backwards compatibility) tests.